core/src/Text/Interpolation/Nyan/Core.hs

```diff
@@ -84,6 +84,11 @@ mkInt iopts = TH.QuasiQuoter
   , TH.quoteDec = \_ ->
       fail "Cannot interpolate at declaration position"
   }
+  where
+    -- as in https://github.com/serokell/nyan-interpolation/issues/21,
+    -- we should use LF line endings even when source file has CRLF
+    useUnixLineEndings :: T.Text -> T.Text
+    useUnixLineEndings = T.replace "\r\n" "\n"
```

Comment:
Better fix this in `Parser.hs` module, since the invariant for Haskell strings is to always carry `\n` as a newline.

This should make the parser simpler too.

Reply:
I agree that parser can be simplified now (so we can delete `\r`-related stuff). But I still think that it's better to remove `\r`'s as soon as possible "since the invariant for Haskell strings is to always carry \n as a newline". So we can think about `quoteExp` input as a raw input, and then give good input to the parser.

Reply:
Oh right, I failed to read what exactly you are doing.

Good, thanks for correcting me.